diff --git a/node_modules/@react-native-async-storage/async-storage/android/build.gradle b/node_modules/@react-native-async-storage/async-storage/android/build.gradle
index 2e4ac94..83fcbee 100644
--- a/node_modules/@react-native-async-storage/async-storage/android/build.gradle
+++ b/node_modules/@react-native-async-storage/async-storage/android/build.gradle
@@ -18,7 +18,6 @@ buildscript {
 
 apply from: "config.gradle"
 apply plugin: "com.android.library"
-apply plugin: "kotlin-android"
 apply plugin: "com.facebook.react"
 
 // ksp/room is required to support legacy storage
diff --git a/node_modules/@react-native-async-storage/async-storage/android/config.gradle b/node_modules/@react-native-async-storage/async-storage/android/config.gradle
index 0b12d2f..a1945d4 100644
--- a/node_modules/@react-native-async-storage/async-storage/android/config.gradle
+++ b/node_modules/@react-native-async-storage/async-storage/android/config.gradle
@@ -30,6 +30,11 @@ String getKspVersion(String kotlinVersion) {
         return overriddenKspVersion
     }
 
+    // TODO: Add guard for AGP 9 and above
+    if (kotlinVersion.startsWith("2.2.")) {
+        return "2.3.8"
+    }
+
     // https://kotlinlang.org/docs/releases.html#release-details
     // https://github.com/google/ksp/releases
     def kspVersions = [
