diff --git a/node_modules/@react-native-async-storage/async-storage/android/build.gradle b/node_modules/@react-native-async-storage/async-storage/android/build.gradle
index 2e4ac94..ae95471 100644
--- a/node_modules/@react-native-async-storage/async-storage/android/build.gradle
+++ b/node_modules/@react-native-async-storage/async-storage/android/build.gradle
@@ -1,6 +1,5 @@
 buildscript {
     apply from: "config.gradle"
-    def kotlinVersion = ext.AsyncStorage.kotlinVersion
     def kspVersion = ext.AsyncStorage.kspVersion
 
     repositories {
@@ -9,8 +8,6 @@ buildscript {
     }
 
     dependencies {
-        classpath "com.android.tools.build:gradle:8.7.2"
-        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:${kotlinVersion}"
         classpath "com.google.devtools.ksp:symbol-processing-gradle-plugin:${kspVersion}"
     }
 }
@@ -18,7 +15,6 @@ buildscript {
 
 apply from: "config.gradle"
 apply plugin: "com.android.library"
-apply plugin: "kotlin-android"
 apply plugin: "com.facebook.react"
 
 // ksp/room is required to support legacy storage
diff --git a/node_modules/@react-native-async-storage/async-storage/android/config.gradle b/node_modules/@react-native-async-storage/async-storage/android/config.gradle
index 0b12d2f..dbad022 100644
--- a/node_modules/@react-native-async-storage/async-storage/android/config.gradle
+++ b/node_modules/@react-native-async-storage/async-storage/android/config.gradle
@@ -30,6 +30,10 @@ String getKspVersion(String kotlinVersion) {
         return overriddenKspVersion
     }
 
+    if (kotlinVersion.startsWith("2.3.")) {
+        return "2.3.8"
+    }
+
     // https://kotlinlang.org/docs/releases.html#release-details
     // https://github.com/google/ksp/releases
     def kspVersions = [
diff --git a/node_modules/@react-native-async-storage/async-storage/android/gradle.properties b/node_modules/@react-native-async-storage/async-storage/android/gradle.properties
index 87fedff..0d35792 100644
--- a/node_modules/@react-native-async-storage/async-storage/android/gradle.properties
+++ b/node_modules/@react-native-async-storage/async-storage/android/gradle.properties
@@ -1,4 +1,4 @@
-AsyncStorage_kotlinVersion=2.2.10
+AsyncStorage_kotlinVersion=2.3.21
 AsyncStorage_roomVersion=2.7.2
 AsyncStorage_minSdkVersion=24
 AsyncStorage_targetSdkVersion=36
