diff --git a/node_modules/@shopify/react-native-skia/android/build.gradle b/node_modules/@shopify/react-native-skia/android/build.gradle
index 437408b..b74ebeb 100644
--- a/node_modules/@shopify/react-native-skia/android/build.gradle
+++ b/node_modules/@shopify/react-native-skia/android/build.gradle
@@ -228,8 +228,10 @@ android {
         ]
     }
 
+
     if (ENABLE_PREFAB) {
         buildFeatures {
+                buildConfig true
                 prefab true
                 prefabPublishing true
         }
