diff --git a/node_modules/react-native-screens/android/build.gradle b/node_modules/react-native-screens/android/build.gradle
index d3ede15..e6b32be 100644
--- a/node_modules/react-native-screens/android/build.gradle
+++ b/node_modules/react-native-screens/android/build.gradle
@@ -86,7 +86,6 @@ if (isRunningInContextOfScreensRepo()) {
     apply plugin: "com.facebook.react"
 }
 apply plugin: 'com.android.library'
-apply plugin: 'kotlin-android'
 
 def reactNativeArchitectures() {
     def value = project.getProperties().get("reactNativeArchitectures")
@@ -175,16 +174,15 @@ android {
     }
     sourceSets.main {
         ext.androidResDir = "src/main/res"
-        java {
-            srcDirs += [
-                "src/fabric/java",
-            ]
+        kotlin {
+            directories.add("src/fabric/java")
         }
         res {
             if (safeExtGet(['compileSdkVersion', 'compileSdk'], rnsDefaultCompileSdkVersion) >= 33) {
-                srcDirs = ["${androidResDir}/base", "${androidResDir}/v33"]
+                directories.add("${androidResDir}/base")
+                directories.add("${androidResDir}/v33")
             } else {
-                srcDirs = ["${androidResDir}/base"]
+                directories.add("${androidResDir}/base")
             }
         }
     }
