diff --git a/node_modules/react-native-vision-camera/android/build.gradle b/node_modules/react-native-vision-camera/android/build.gradle
index e6c990f..c1b7aeb 100644
--- a/node_modules/react-native-vision-camera/android/build.gradle
+++ b/node_modules/react-native-vision-camera/android/build.gradle
@@ -19,7 +19,6 @@ def isNewArchitectureEnabled() {
 }
 
 apply plugin: "com.android.library"
-apply plugin: 'org.jetbrains.kotlin.android'
 apply from: '../nitrogen/generated/android/VisionCamera+autolinking.gradle'
 apply from: "./fix-prefab.gradle"
 
diff --git a/node_modules/react-native-vision-camera/android/fix-prefab.gradle b/node_modules/react-native-vision-camera/android/fix-prefab.gradle
index d6c010e..57513a0 100644
--- a/node_modules/react-native-vision-camera/android/fix-prefab.gradle
+++ b/node_modules/react-native-vision-camera/android/fix-prefab.gradle
@@ -12,39 +12,20 @@ tasks.configureEach { task ->
   }
 }
 
-afterEvaluate {
-  def abis = reactNativeArchitectures()
-  rootProject.allprojects.each { proj ->
-    if (proj === rootProject) return
-
-    def dependsOnThisLib = proj.configurations.findAll { it.canBeResolved }.any { config ->
-      config.dependencies.any { dep ->
-        dep.group == project.group && dep.name == project.name
+androidComponents {
+  onVariants(selector().all()) { variant ->
+    def abis = reactNativeArchitectures()
+    def variantName = variant.name
+    abis.each { abi ->
+      def searchDir = new File(project.projectDir, ".cxx/${variantName}")
+      if (!searchDir.exists()) return
+      def matches = []
+      searchDir.eachDir { randomDir ->
+        def prefabFile = new File(randomDir, "${abi}/prefab_config.json")
+        if (prefabFile.exists()) matches << prefabFile
       }
-    }
-    if (!dependsOnThisLib && proj != project) return
-
-    if (!proj.plugins.hasPlugin('com.android.application') && !proj.plugins.hasPlugin('com.android.library')) {
-      return
-    }
-
-    def variants = proj.android.hasProperty('applicationVariants') ? proj.android.applicationVariants : proj.android.libraryVariants
-    // Touch the prefab_config.json files to ensure that in ExternalNativeJsonGenerator.kt we will re-trigger the prefab CLI to
-    // generate a libnameConfig.cmake file that will contain our native library (.so).
-    // See this condition: https://cs.android.com/android-studio/platform/tools/base/+/mirror-goog-studio-main:build-system/gradle-core/src/main/java/com/android/build/gradle/tasks/ExternalNativeJsonGenerator.kt;l=207-219?q=createPrefabBuildSystemGlue
-    variants.all { variant ->
-      def variantName = variant.name
-      abis.each { abi ->
-        def searchDir = new File(proj.projectDir, ".cxx/${variantName}")
-        if (!searchDir.exists()) return
-        def matches = []
-        searchDir.eachDir { randomDir ->
-          def prefabFile = new File(randomDir, "${abi}/prefab_config.json")
-          if (prefabFile.exists()) matches << prefabFile
-        }
-        matches.each { prefabConfig ->
-          prefabConfig.setLastModified(System.currentTimeMillis())
-        }
+      matches.each { prefabConfig ->
+        prefabConfig.setLastModified(System.currentTimeMillis())
       }
     }
   }
diff --git a/node_modules/react-native-vision-camera/nitrogen/generated/android/VisionCamera+autolinking.gradle b/node_modules/react-native-vision-camera/nitrogen/generated/android/VisionCamera+autolinking.gradle
index 5f239bb..93f62e0 100644
--- a/node_modules/react-native-vision-camera/nitrogen/generated/android/VisionCamera+autolinking.gradle
+++ b/node_modules/react-native-vision-camera/nitrogen/generated/android/VisionCamera+autolinking.gradle
@@ -18,10 +18,8 @@ logger.warn("[NitroModules] 🔥 VisionCamera is boosted by nitro!")
 android {
   sourceSets {
     main {
-      java.srcDirs += [
-        // Nitrogen files
-        "${project.projectDir}/../nitrogen/generated/android/kotlin"
-      ]
+      // Nitrogen files
+      kotlin.directories.add("${project.projectDir}/../nitrogen/generated/android/kotlin")
     }
   }
 }
