diff --git a/node_modules/react-native-safe-area-context/android/build.gradle b/node_modules/react-native-safe-area-context/android/build.gradle
index f324b9a..275d1ea 100644
--- a/node_modules/react-native-safe-area-context/android/build.gradle
+++ b/node_modules/react-native-safe-area-context/android/build.gradle
@@ -7,8 +7,6 @@ buildscript {
     }
 
     dependencies {
-        classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version")
-        classpath("com.android.tools.build:gradle:7.3.1")
         classpath("com.diffplug.spotless:spotless-plugin-gradle:6.11.0")
     }
 }
@@ -32,7 +30,6 @@ def isNewArchitectureEnabled() {
 }
 
 apply plugin: 'com.android.library'
-apply plugin: 'kotlin-android'
 
 if (isNewArchitectureEnabled()) {
     apply plugin: "com.facebook.react"
@@ -82,17 +79,20 @@ android {
         exclude "**/libreact_render*.so"
     }
     sourceSets.main {
-        java {
-            if (isNewArchitectureEnabled()) {
-                srcDirs += [
-                    "src/fabric/java",
-                    "${project.buildDir}/generated/source/codegen/java"
-                ]
-            } else {
-                srcDirs += [
-                    "src/paper/java"
-                ]
-            }
+        if (isNewArchitectureEnabled()) {
+            java.srcDirs += [
+                "${project.buildDir}/generated/source/codegen/java" // contains only java files
+            ]
+            kotlin.srcDirs += [
+                "src/fabric/java", // contains only kotlin files
+            ]
+        } else {
+            java.srcDirs += [
+                "src/paper/java" // contains both java and kotlin files
+            ]
+            kotlin.srcDirs += [
+                "src/paper/java" // contains both java and kotlin files
+            ]
         }
     }
 }
